app/presenters/content_type_resolver.rb

```diff
@@ -22,23 +22,17 @@ def resolve_hash(hash)
   end
 
   def resolve_array(array)
-    if has_content_types?(array)
-      extract_content(array)
-    else
-      array.map { |e| resolve(e) }
-    end
-  end
+    if array.all? { |v| v.is_a?(Hash) }
+      array = array.map(&:symbolize_keys)
 
-  def has_content_types?(array)
-    return false unless array.all? { |v| v.is_a?(Hash) }
+      content_for_type = array.detect do |h|
+        h[:content_type] == content_type && h[:content]
+      end
 
-    hash_keys = array.flat_map(&:keys).map(&:to_sym)
-    hash_keys.include?(:content_type) && hash_keys.include?(:content)
-  end
+      return content_for_type[:content] if content_for_type
```

Comment:
This is returning the original value if the `content_for_type` condition doesn't pass because of the `array.detect`. It might be worth adding an explicit `return array` after this line so the behaviour is more obvious.